components/blocks/dynamicCardGridBlock.tsx

```diff
@@ -0,0 +1,172 @@
+import classNames from "classnames";
+import Image from "next/image";
+import { Key } from "react";
+import { Template } from "tinacms";
+import { CustomLink } from "../customLink";
+import { isEmpty } from "../training/eventBooking";
+
+export type DynamicCardGridBlockProps = {
+  gridLayout?: keyof typeof GridCols;
+  imageList: ImgCard[];
+};
+
+export type ImgCard = {
+  title: string;
+  subTitle?: string;
+  link?: {
+    url: string;
+    label: string;
+  };
+  altText: string;
+  imageSrc: string;
+  height?: number;
+  width?: number;
+};
+
+const GridCols = {
+  single: "col-span-12",
+  double: "md:col-span-6",
+  triple: "lg:col-span-6 xl:col-span-4",
+  four: "md:col-span-6 lg:col-span-4 xl:col-span-3",
+} as const;
+
+export const DynamicCardGridBlock: React.FC<DynamicCardGridBlockProps> = ({
+  imageList,
+  gridLayout,
+}) => {
+  return (
+    <div className="grid grid-cols-12 gap-5 py-1">
+      {imageList?.map((imageblock, index: Key) => {
+        return <Card card={imageblock} gridLayout={gridLayout} key={index} />;
+      })}
+    </div>
+  );
+};
+
+interface CardProps {
+  card: ImgCard;
+  gridLayout: keyof typeof GridCols;
+}
+
+const Card = ({ card, gridLayout }: CardProps) => {
+  const { title, subTitle, imageSrc, altText, height, width, link } = card;
+  return (
+    <div
+      className={classNames(
+        "relative col-span-12 flex flex-col justify-between rounded border-1 text-center",
+        GridCols[gridLayout ?? "triple"]
+      )}
+    >
+      {imageSrc && (
+        <div className="flex grow justify-center py-4">
+          <Image
+            src={imageSrc}
+            alt={altText ?? ""}
+            height={isEmpty(height) ? 150 : height} // IsEmpty - Tina converts height to empty string after removing the text from the field
+            width={isEmpty(width) ? 150 : width} // IsEmpty - Tina converts width to empty string after removing the text from the field
```

Comment:
isn't an empty string falsey (i.e. you don't need this function and could just use `||`)

Reply:
This was causing an issue. When a user cleared the width field, Tina broke the page because a number field cannot accept an empty string.